templates/15-rust.sh

```diff
@@ -94,15 +92,21 @@ build_rust_build() {
 			|| abdie "Compilation failed: $?."
 	else
 		abinfo 'Using fallback build method ...'
-	  if [[ "${CARGO_AFTER[@]}" =~ '-p' ]]; then
-		workspace_flag=''
-	  fi
-	  cargo build "${DEFAULT_CARGO_CONFIG[@]}" \
+		# determine if buiding a package
+		local workspace_flag
+		workspace_flag="--workspace"
+		for element in "${CARGO_AFTER[@]}"; do
+			if [[ "$element" == "-p" || "$element" == "--package" ]]; then
```

Comment:
Okay, so the issue was really that we didn't account for the full form of the `-p` flag.

Your commit message is then inaccurate. It should be:

Account for the `--package` flag in the workspace-flag-unset logic, as it is the full form for `-p`.

Reply:
Then...

Why not just keep using the original if branch? Why the for loop?

Reply:
I have discussed this problem with @eatradish . Just imagine that we have a package with CARGO_AFTER="--features no-parallel" (or any other arguments that happen to contain "-p". Are we building a package in a workspace? Definitely not.

Reply:
> I have discussed this problem with @eatradish . Just imagine that we have a package with CARGO_AFTER="--features no-parallel" (or any other arguments that happen to contain "-p". Are we building a package in a workspace? Definitely not.

Sure, then explain both in the commit message, my point still stands (with an additional case that you have mentioned).

The for loop is still unnecessary.